server/test/services/program/ProgramServiceTest.java

```diff
@@ -1063,6 +1070,31 @@ public void getProgramDefinitionAsync_constructsQuestionDefinitions() throws Exc
     assertThat(foundQuestion).isInstanceOf(NameQuestionDefinition.class);
   }
 
+  @Test
+  public void getProgramDefinitionAsync_setsCacheWithQuestionDefinitions() throws Exception {
+    QuestionDefinition question = nameQuestion;
+    ProgramDefinition program =
+        ProgramBuilder.newActiveProgram()
```

Comment:
Question (non-blocking): Why is this test able to use `newActiveProgram` but other tests needed to be changed to use `newDraftProgram`?

Reply:
Good question! Basically, we should set it as a draft if we're still planning to make changes to the program (which the other tests do) and we don't want to cache it. In this case, we don't make changes after creating the program and we want it to be cached, so we can set it as active. I added a test case as well to ensure if its a draft program, it doesn't get cached.

Reply:
Ah got it! Does the `.withBlock().withRequiredQuestionDefinition(question)` not count as making changes after creating the program?

Reply:
It'll all get built as one I think and the program def won't be cached until we make the call to get the program definition.